File: src/se/clau/intellijlux/psi/LuxElementFactory.java
package se.clau.intellijlux.psi;

import com.intellij.openapi.project.Project;
import com.intellij.psi.*;
import se.clau.intellijlux.LuxFile;
import se.clau.intellijlux.LuxFileType;

public class LuxElementFactory {
  public static LuxIdent createProperty(Project project, String name) {
    final LuxFile file = createFile(project, name);
    return (LuxIdent) file.getFirstChild();
  }

  public static LuxFile createFile(Project project, String text) {
    String name = "dummy.lux";
    return (LuxFile) PsiFileFactory.getInstance(project).
            createFileFromText(name, LuxFileType.INSTANCE, text);
  }
}

File: src/se/clau/intellijlux/psi/LuxElementFactory.kt
package se.clau.intellijlux.psi

import com.intellij.openapi.project.Project
import com.intellij.psi.PsiFileFactory
import se.clau.intellijlux.LuxFile
import se.clau.intellijlux.LuxFileType

object LuxElementFactory {
  fun createProperty(project: Project?, name: String?): LuxIdent {
    val file = createFile(project, name)
    return file.firstChild as LuxIdent
  }

  fun createFile(project: Project?, text: String?): LuxFile {
    val name = "dummy.lux"
    return PsiFileFactory.getInstance(project)
      .createFileFromText(name, LuxFileType, text!!) as LuxFile
  }
}